### PR TITLE
Python parser can now indentify triple-quoted string comments

### PR DESCRIPTION
**Pull request by itiel**
Python comment parser can now indentify two kinds of comments using the tokenize module.
- Single-lined comments which begin with the `#` character and end with a line-break.
- Multi-lined comments or docstrings, which are just triple-quoted strings (start and end with `'''` or `"""`), are told apart from regular strings by the type of the previous token which should be a line-break or an indentation (`NEWLINE`, `NL`, `INDENT` or `DEDENT`) or no token at all (it would mean it's the fisrt thing in the script). Even in cases like this:

      my_string = \
      '''this should not be considered a comment'''
      my_string = \
        '''this should not either''' # <- notice the increasing indentation
      my_string = \
          '''weird syntax anyway''' # <- but still valid indentation

  the previous token to the string is the `=` operator and not a line-break or an indentation. That way, only triple-quoted strings preceded by a line-break, an indentation, or no token, will be considered intended as comments.

This solves issue https://github.com/jeanralphaviles/comment_parser/issues/33

### CODE REVIEW

**Comment by tim-puhlfuerss** [comment]
Hi, your solution does not work for triple-quoted comments in the beginning of files (at least not for the ones that are not the first file). 
However, it works when you add `tokenize.ENCODING` to `multicommprevnums`.

File: comment_parser/parsers/python_parser.py [hunk 1]
```diff
@@ -5,12 +5,29 @@
...
   """Extracts a list of comments from the given Python script.
 
-  Comments are identified using the tokenize module. Does not include function,
-  class, or module docstrings. All comments are single line comments.
+  Comments are identified using the tokenize module.
+    - Single-lined comments which begin with the '#' character and end with a line-break.
+    - Multi-lined comments or docstrings, which are just triple-quoted strings (start 
+      and end with ''' or 3 of these "), are told apart from regular strings by the 
+      type of the previous token which should be a line-break or an indentation (NEWLINE, 
+      NL, INDENT or DEDENT) or no token at all (it would mean it's the fisrt thing in
```
Comment (jeanralphaviles) [inline comment_parser/parsers/python_parser.py:16]: s/fisrt/first/g

File: comment_parser/parsers/python_parser.py [hunk 2]
```diff
@@ -5,12 +5,29 @@
...
+
+        my_string = \
+          '''this should not either''' # <- notice the increasing indentation
+
+        my_string = \
+            '''weird syntax anyway''' # <- but still valid indentation
+      
+      the previous token to the string is the '=' operator and not a line-break or an  
+      indentation. That way, only triple-quoted strings preceded by a line-break, an 
+      indentation, or no token, will be considered intended as comments.
```
Comment (jeanralphaviles) [inline comment_parser/parsers/python_parser.py:30]: Drop "intended as"

File: comment_parser/parsers/python_parser.py [hunk 3]
```diff
@@ -19,11 +36,24 @@ def extract_comments(code):
   Raises:
     tokenize.TokenError
   """
+  triplequotes = ['"""', "'''"]
+  multicommprevnums = [tokenize.NEWLINE, tokenize.NL, tokenize.INDENT, tokenize.DEDENT]
```
Comment (jeanralphaviles) [inline comment_parser/parsers/python_parser.py:40]: Could we call this something like "breaks" instead? multicommprevnums is quite a long variable name.

File: comment_parser/parsers/python_parser.py [hunk 4]
```diff
@@ -19,11 +36,24 @@ def extract_comments(code):
...
+  prevtoknum = None # Stores the previous token's type.
```
Comment (jeanralphaviles) [inline comment_parser/parsers/python_parser.py:41]: Just prevtok?

File: comment_parser/parsers/python_parser.py [hunk 5]
```diff
@@ -5,12 +5,29 @@
...
-  class, or module docstrings. All comments are single line comments.
+  Comments are identified using the tokenize module.
+    - Single-lined comments which begin with the '#' character and end with a line-break.
+    - Multi-lined comments or docstrings, which are just triple-quoted strings (start 
+      and end with ''' or 3 of these "), are told apart from regular strings by the 
+      type of the previous token which should be a line-break or an indentation (NEWLINE, 
+      NL, INDENT or DEDENT) or no token at all (it would mean it's the fisrt thing in 
+      the script). Even in cases like this: 
+        
+        my_string = \
```
Comment (jeanralphaviles) [inline comment_parser/parsers/python_parser.py:19]: Can you add test cases covering these examples?

File: comment_parser/parsers/python_parser.py [hunk 6]
```diff
@@ -19,11 +36,24 @@ def extract_comments(code):
...
   for toknum, tokstring, tokloc, _, _ in tokens:
+    # Single-lined comment.
     if toknum is tokenize.COMMENT:
       # Removes leading '#' character.
       tokstring = tokstring[1:]
       comments.append(common.Comment(tokstring, tokloc[0], False))
+      continue
+    # Multi-lined comment.
+    if toknum is tokenize.STRING:
+      if tokstring[:3] in triplequotes and tokstring[-3:] in triplequotes:
```
Comment (jeanralphaviles) [inline comment_parser/parsers/python_parser.py:53]: Please add a test for an empty single-line comment. For example: #. I suspect that the [:3] and [-3:] will cause an array exception.

File: comment_parser/parsers/python_parser.py [hunk 7]
```diff
@@ -19,11 +36,24 @@ def extract_comments(code):
...
+        if (not prevtoknum) or prevtoknum in multicommprevnums:
+          # Removes the leading and preceding 3ple quotes (""" or ''').
```
Comment (jeanralphaviles) [inline comment_parser/parsers/python_parser.py:55]: Prefer "triple" to "3ple" as it's more straightforward.

File: comment_parser/parsers/python_parser.py [hunk 8]
```diff
@@ -19,11 +36,24 @@ def extract_comments(code):
...
+  prevtoknum = None # Stores the previous token's type.
   comments = []
   tokens = tokenize.tokenize(io.BytesIO(code.encode()).readline)
   for toknum, tokstring, tokloc, _, _ in tokens:
+    # Single-lined comment.
     if toknum is tokenize.COMMENT:
       # Removes leading '#' character.
       tokstring = tokstring[1:]
       comments.append(common.Comment(tokstring, tokloc[0], False))
+      continue
```
Comment (jeanralphaviles) [inline comment_parser/parsers/python_parser.py:50]: This continue statement is unnecessary. If we fall through the preceeding if statement, we won't enter the next one.
